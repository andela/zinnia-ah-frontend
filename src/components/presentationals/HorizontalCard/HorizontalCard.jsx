import React from 'react';

import PopularAuthorItem from '../PopularAuthorItem/PopularAuthorItem';

// components

// styles
import './HorizontalCard.scss';
import Title from '../Title/Title';
import Image from '../Image/Image';

const HorizontalCard = () => (
  <div className="h-card">
    <div>
      <Image
        src="https://res.cloudinary.com/nedy123/image/upload/v1514128720/mike_vobxlw.jpg"
        className="h-card-image"
      />
    </div>
    <div
      className="content"
      style={{
        paddingLeft: '1rem',
      }}
    >
      <Title className="title-md mb-0" content="Nature and Life" />
      <p className="description">
        A tab could be a different number of columns depending on your
        environment, but a space is always one column.
      </p>
      <PopularAuthorItem
        key={'uiojklnmkopiuygh'}
        name={'ebenezer'}
        url={'notontwitter.com'}
        image={''}
        username={'eben'}
      />
    </div>
  </div>
);

export default HorizontalCard;
