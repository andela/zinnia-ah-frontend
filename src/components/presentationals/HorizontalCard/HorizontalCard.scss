.h-card {
  display: flex;
  max-height: 11.575rem;
  width: 100%;

  .h-card-image {
    height: 100%;
    width: 12.5rem;
  }

  .content {
    width: 100%;
  }
}

.mb-0 {
  margin-bottom: .2rem;
}

.h-card:nth-child(1) {
  margin-bottom: 2.8125rem;
}
