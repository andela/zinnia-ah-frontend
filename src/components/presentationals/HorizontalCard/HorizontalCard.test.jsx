import React from 'react';
import { mount } from 'enzyme';

import HorizontalCard from './HorizontalCard';

describe('Horizontal Card', () => {
  it('should render correctly', () => {
    const component = mount(<HorizontalCard className="" />);
    expect(component).toMatchSnapshot();
  });
});
