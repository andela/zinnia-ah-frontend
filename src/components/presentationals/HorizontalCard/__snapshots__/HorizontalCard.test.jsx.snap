// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Horizontal Card should render correctly 1`] = `
<HorizontalCard
  className=""
>
  <div
    className="h-card"
  >
    <div>
      <Image
        className="h-card-image"
        src="https://res.cloudinary.com/nedy123/image/upload/v1514128720/mike_vobxlw.jpg"
      >
        <img
          className="h-card-image"
          src="https://res.cloudinary.com/nedy123/image/upload/v1514128720/mike_vobxlw.jpg"
        />
      </Image>
    </div>
    <div
      className="content"
      style={
        Object {
          "paddingLeft": "1rem",
        }
      }
    >
      <Title
        className="title-md mb-0"
        content="Nature and Life"
      >
        <h1
          className="title title-md mb-0"
        >
          Nature and Life
        </h1>
      </Title>
      <p
        className="description"
      >
        A tab could be a different number of columns depending on your environment, but a space is always one column.
      </p>
      <PopularAuthorItem
        image=""
        key="uiojklnmkopiuygh"
        name="ebenezer"
        url="notontwitter.com"
        username="eben"
      >
        <div
          className="author-slot"
        >
          <UserThumbnail
            href="#"
            image=""
            name="ebenezer"
            tag="a"
            url="notontwitter.com"
            username="eben"
          >
            <a
              className="user-thumbnail"
              href="notontwitter.com"
            >
              <Avatar
                className=""
                url="https://res.cloudinary.com/nedy123/image/upload/v1557149386/avatar_pupqrh.svg"
              >
                <img
                  alt="avatar"
                  className="avatar "
                  src="https://res.cloudinary.com/nedy123/image/upload/v1557149386/avatar_pupqrh.svg"
                />
              </Avatar>
              <div>
                <span
                  className="name"
                >
                   
                  ebenezer
                   
                </span>
                <span
                  className="info"
                >
                   @
                  eben
                   
                </span>
              </div>
            </a>
          </UserThumbnail>
          <Button
            className="btn-white right-small"
            type="submit"
            value="See Posts"
          >
            <button
              className="btn btn-white right-small"
              type="submit"
            >
              See Posts
            </button>
          </Button>
        </div>
      </PopularAuthorItem>
    </div>
  </div>
</HorizontalCard>
`;
